node/GUIDE.md

```diff
@@ -0,0 +1,230 @@
+# Setting up a new node service with TChannel, Thrift & Hyperbahn
+
+This guide will show you how to write a server and client.
+
+The code matching this guide is [here](./examples/keyvalue/).
+
+## Dependencies
+
+Make sure you have `node` and `npm` installed. To get started
+create an empty project with `npm init` and run
+`npm install tchannel --save`.
+
+## Defining your service
+
+To define the interface of your service you can create a
+[Thrift](https://thrift.apache.org) file.
+
+Today we will use:
+
+`keyvalue.thrift`:
+```thrift
+struct GetResult {
+    1: string value
+}
+
+service KeyValue {
+    GetResult get_v1(
+        1: string key
+    )
+    void put_v1(
+        1: string key,
+        2: string value
+    )
+}
+```
+
+Our service will have two tchannel endpoints that can be.
+Namely `KeyValue::get_v1` and `KeyValue::put_v1`.
```

Comment:
I'd like to recommend encoding the version name somewhere. Putting it in the method name allows us to change the version of one method at a time with maximum granularity.

The idea that you "simply never break back-compat" is a valid idea but thats not how the real world works.